File: src/01-easy/lumen/solution.pas
program Answer;
uses SysUtils, Math;
var n,l,r,c,r2,c2,d,idx:integer;
    g:array[0..49,0..49] of char;
    line:string;
    lit:boolean;
begin
    readln(n);readln(l);
    for r:=0 to n-1 do begin
        readln(line); line:=Trim(line); idx:=0;
        for c:=0 to n-1 do begin
            while (idx<Length(line)) and (line[idx+1]=' ') do Inc(idx);
            g[r,c]:=line[idx+1]; Inc(idx,2);
        end;
    end;
    d:=0;
    for r:=0 to n-1 do for c:=0 to n-1 do begin
        lit:=false;
        for r2:=0 to n-1 do for c2:=0 to n-1 do
            if(g[r2,c2]='C')and(Max(Abs(r-r2),Abs(c-c2))<l)then lit:=true;
        if not lit then Inc(d);
    end;
    writeln(d);
end.

File: src/01-easy/lumen/solution.ml
let () =
  let n = int_of_string (String.trim (input_line stdin)) in
  let l = int_of_string (String.trim (input_line stdin)) in
  let g = Array.init n (fun _ ->
    let line = String.trim (input_line stdin) in
    let parts = String.split_on_char ' ' line in
    Array.of_list parts) in
  let d = ref 0 in
  for r = 0 to n-1 do for c = 0 to n-1 do
    let lit = ref false in
    for r2 = 0 to n-1 do for c2 = 0 to n-1 do
      if g.(r2).(c2) = "C" && max (abs(r-r2)) (abs(c-c2)) < l then lit := true
    done done;
    if not !lit then incr d
  done done;
  print_int !d; print_newline ()